### PR TITLE
Version checker: changed QRegExp to  QRegularExpression

### PR DESCRIPTION
**Pull request by DzianCPP**
Refactored parseLatestReleasedVersion(); so that is used QRegularExpression instead of not supported QRegExp in Qt6.

### CODE REVIEW

File: src/alz_serializer.cpp [hunk 1]
```diff
@@ -465,7 +465,8 @@ std::unique_ptr<MindMapData> fromXml(QDomDocument document)
                               font.setOverline(e.attribute(DataKeywords::MindMap::FONT_OVERLINE).toInt());
                               font.setUnderline(e.attribute(DataKeywords::MindMap::FONT_UNDERLINE).toInt());
                               font.setStrikeOut(e.attribute(DataKeywords::MindMap::FONT_STRIKE_OUT).toInt());
-                              font.setWeight(e.attribute(DataKeywords::MindMap::FONT_WEIGHT).toInt());
+                              font.setWeight(QFont::Weight::Bold);
+                              //font.setWeight(e.attribute(DataKeywords::MindMap::FONT_WEIGHT).toInt());
```
Comment (juzzlin) [inline src/alz_serializer.cpp:469]: Unrelated changes and commented code.

File: src/version_checker.cpp [hunk 2]
```diff
@@ -32,9 +33,9 @@ Version parseLatestReleasedVersion(QString data)
 {
     Version newest;
     for (auto && line : data.split("\n")) {
-        QRegExp tagMatch("^*tag/(\\d.\\d.\\d).*$");
-        if (const auto index = tagMatch.indexIn(line); index != -1) {
-            const Version version(tagMatch.cap(1));
+        QRegularExpression tagMatch("\\d{1,3}.\\d.\\d?");
```
Comment (juzzlin) [inline src/version_checker.cpp:36]: How can you be sure this matches strings like "releases/tag/3.6.0" when you remove "^*tag/" ?

**Review by juzzlin — CHANGES_REQUESTED** [review]
Good change, but the commit should change only what it claims. 

Also, I'm not sure the new expression always works as it looks like less explicit.

**Comment by DzianCPP** [comment]
Got it, will re-do the things.
